src/aiotaskq/task.py

```diff
@@ -27,24 +31,42 @@ class AsyncResult(t.Generic[RT]):
     """
 
     pubsub: IPubSub
-    _result: RT
-    _completed: bool = False
-    _task_id: str
-
-    def __init__(self, task_id: str) -> None:
+    task_id: str
+    ready: bool = False
+    result: RT | None
+    error: Exception | None
+
+    def __init__(
+        self, task_id: str, result: RT | None, ready: bool, error: Exception | None
+    ) -> None:
         """Store task_id in AsyncResult instance."""
-        self._task_id = task_id
-        self.pubsub = PubSub.get(url=REDIS_URL, poll_interval_s=0.01)
+        self.task_id = task_id
+        self.ready = ready
+        self.result = result
+        self.error = error
+        self.pubsub = PubSub.get(url=Config.broker_url(), poll_interval_s=0.01)
+
+    @classmethod
+    async def from_publisher(cls, task_id: str) -> "AsyncResult":
```

Comment:
I feel like we should move this polling logic elsewhere & keep `AsyncResult` as a simple result definition

Reply:
Moved logic from `AsyncResult.from_publisher` to `Task._get_result`, and removed `AsyncResult.from_publisher`. Now is a simple result definition.